static/src/javascripts/projects/common/modules/identity/sign-in-gate/variants/centesimus/control.js

```diff
@@ -18,16 +19,18 @@ const variant = 'centesimus-control-1';
 
 // method which returns a boolean determining if this variant can be shown on the current pageview
 const canShow: (name?: string) => boolean = (name = '') =>
-    !hasUserDismissedGate({
-        name,
-        variant,
-        componentName,
-    }) &&
-    isNPageOrHigherPageView(3) &&
-    !isLoggedIn() &&
-    !isInvalidArticleType() &&
-    !isInvalidSection() &&
-    !isIOS9();
+    setIsGateRendered(
+        !hasUserDismissedGate({
+            name,
+            variant,
+            componentName,
+        }) &&
+            isNPageOrHigherPageView(3) &&
+            !isLoggedIn() &&
+            !isInvalidArticleType() &&
+            !isInvalidSection() &&
+            !isIOS9()
+    );
```

Comment:
instead of passing in all the checks to `setIsGateRendered` and having that return the boolean, it would be better to separate the concerns by putting all the checks into a variable (e.g. `const canShowCheck = !hasUser..` etc) and passing that into `setIsGateRendered` instead, then just returning `canShowCheck` at the end of the `canShow` method.